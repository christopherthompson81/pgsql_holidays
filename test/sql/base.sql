BEGIN;

CREATE EXTENSION holidays;

SELECT * FROM holidays.by_country ('FR'::text, 2022, 2023);

SELECT * FROM holidays.by_country ('FR'::text, '10/01/2022'::date, '05/31/2023'::date);

SELECT * FROM holidays.by_country ('FR'::text, daterange('2022/10/01', '2023/05/31'));

ROLLBACK;
