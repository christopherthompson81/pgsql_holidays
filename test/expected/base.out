BEGIN;
CREATE EXTENSION holidays;
SELECT * FROM holidays.by_country ('FR'::text, 2022, 2023);
 datestamp  |       reference       |    description     | authority | day_off | observation_shifted | start_time | end_time 
------------+-----------------------+--------------------+-----------+---------+---------------------+------------+----------
 01-01-2022 | New Year's Day        | Jour de l'an       | national  | t       | f                   | 00:00:00   | 24:00:00
 05-01-2022 | Labour Day            | Fête du Travail    | national  | t       | f                   | 00:00:00   | 24:00:00
 05-08-2022 | Victory in Europe Day | Armistice 1945     | national  | t       | f                   | 00:00:00   | 24:00:00
 07-14-2022 | Bastille Day          | Fête nationale     | national  | t       | f                   | 00:00:00   | 24:00:00
 11-11-2022 | Remembrance Day       | Armistice 1918     | national  | t       | f                   | 00:00:00   | 24:00:00
 04-18-2022 | Easter Monday         | Lundi de Pâques    | national  | t       | f                   | 00:00:00   | 24:00:00
 06-06-2022 | Pentecost             | Lundi de Pentecôte | national  | t       | f                   | 00:00:00   | 24:00:00
 05-26-2022 | Ascension             | Ascension          | national  | t       | f                   | 00:00:00   | 24:00:00
 08-15-2022 | Assumption            | Assomption         | national  | t       | f                   | 00:00:00   | 24:00:00
 11-01-2022 | All Saints' Day       | Toussaint          | national  | t       | f                   | 00:00:00   | 24:00:00
 12-25-2022 | Christmas Day         | Noël               | national  | t       | f                   | 00:00:00   | 24:00:00
 01-01-2023 | New Year's Day        | Jour de l'an       | national  | t       | f                   | 00:00:00   | 24:00:00
 05-01-2023 | Labour Day            | Fête du Travail    | national  | t       | f                   | 00:00:00   | 24:00:00
 05-08-2023 | Victory in Europe Day | Armistice 1945     | national  | t       | f                   | 00:00:00   | 24:00:00
 07-14-2023 | Bastille Day          | Fête nationale     | national  | t       | f                   | 00:00:00   | 24:00:00
 11-11-2023 | Remembrance Day       | Armistice 1918     | national  | t       | f                   | 00:00:00   | 24:00:00
 04-10-2023 | Easter Monday         | Lundi de Pâques    | national  | t       | f                   | 00:00:00   | 24:00:00
 05-29-2023 | Pentecost             | Lundi de Pentecôte | national  | t       | f                   | 00:00:00   | 24:00:00
 05-18-2023 | Ascension             | Ascension          | national  | t       | f                   | 00:00:00   | 24:00:00
 08-15-2023 | Assumption            | Assomption         | national  | t       | f                   | 00:00:00   | 24:00:00
 11-01-2023 | All Saints' Day       | Toussaint          | national  | t       | f                   | 00:00:00   | 24:00:00
 12-25-2023 | Christmas Day         | Noël               | national  | t       | f                   | 00:00:00   | 24:00:00
(22 rows)

SELECT * FROM holidays.by_country ('FR'::text, '10/01/2022'::date, '05/31/2023'::date);
 datestamp  |       reference       |    description     | authority | day_off | observation_shifted | start_time | end_time 
------------+-----------------------+--------------------+-----------+---------+---------------------+------------+----------
 11-11-2022 | Remembrance Day       | Armistice 1918     | national  | t       | f                   | 00:00:00   | 24:00:00
 11-01-2022 | All Saints' Day       | Toussaint          | national  | t       | f                   | 00:00:00   | 24:00:00
 12-25-2022 | Christmas Day         | Noël               | national  | t       | f                   | 00:00:00   | 24:00:00
 01-01-2023 | New Year's Day        | Jour de l'an       | national  | t       | f                   | 00:00:00   | 24:00:00
 05-01-2023 | Labour Day            | Fête du Travail    | national  | t       | f                   | 00:00:00   | 24:00:00
 05-08-2023 | Victory in Europe Day | Armistice 1945     | national  | t       | f                   | 00:00:00   | 24:00:00
 04-10-2023 | Easter Monday         | Lundi de Pâques    | national  | t       | f                   | 00:00:00   | 24:00:00
 05-29-2023 | Pentecost             | Lundi de Pentecôte | national  | t       | f                   | 00:00:00   | 24:00:00
 05-18-2023 | Ascension             | Ascension          | national  | t       | f                   | 00:00:00   | 24:00:00
(9 rows)

SELECT * FROM holidays.by_country ('FR'::text, daterange('2022/10/01', '2023/05/31'));
 datestamp  |       reference       |    description     | authority | day_off | observation_shifted | start_time | end_time 
------------+-----------------------+--------------------+-----------+---------+---------------------+------------+----------
 11-11-2022 | Remembrance Day       | Armistice 1918     | national  | t       | f                   | 00:00:00   | 24:00:00
 11-01-2022 | All Saints' Day       | Toussaint          | national  | t       | f                   | 00:00:00   | 24:00:00
 12-25-2022 | Christmas Day         | Noël               | national  | t       | f                   | 00:00:00   | 24:00:00
 01-01-2023 | New Year's Day        | Jour de l'an       | national  | t       | f                   | 00:00:00   | 24:00:00
 05-01-2023 | Labour Day            | Fête du Travail    | national  | t       | f                   | 00:00:00   | 24:00:00
 05-08-2023 | Victory in Europe Day | Armistice 1945     | national  | t       | f                   | 00:00:00   | 24:00:00
 04-10-2023 | Easter Monday         | Lundi de Pâques    | national  | t       | f                   | 00:00:00   | 24:00:00
 05-29-2023 | Pentecost             | Lundi de Pentecôte | national  | t       | f                   | 00:00:00   | 24:00:00
 05-18-2023 | Ascension             | Ascension          | national  | t       | f                   | 00:00:00   | 24:00:00
(9 rows)

ROLLBACK;
